bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationRemote.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.argoclima.internal.configuration;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Duration;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.argoclima.internal.ArgoClimaBindingConstants;
+import org.openhab.binding.argoclima.internal.exception.ArgoConfigurationException;
+
+/**
+ * The {@link ArgoClimaConfigurationRemote} class contains fields mapping thing configuration parameters
+ * for a remote Argo device (comms via Argo servers)
+ *
+ * @author Mateusz Bronk - Initial contribution
+ */
+@NonNullByDefault
+public class ArgoClimaConfigurationRemote extends ArgoClimaConfigurationBase {
+
+    /**
+     * The duration after which the device would be considered non-responsive (and taken OFFLINE)
+     */
+    public static final Duration LAST_SEEN_UNAVAILABILITY_THRESHOLD = Duration.ofMinutes(20);
+
+    /**
+     * Argo configuration parameters specific to remote connection
+     * These names are defined in thing-types.xml and get injected on instantiation
+     * through {@link org.openhab.core.thing.binding.BaseThingHandler#getConfigAs getConfigAs}
+     */
+    private String username = "";
+    private String password = "";
+
+    /**
+     * Get the username (login) to use in authenticating to Argo server
+     *
+     * @return username (as configured by the user)
+     */
+    public String getUsername() {
+        return this.username;
+    }
+
+    /**
+     * Get the masked password used in authenticating to Argo server (for logging)
+     *
+     * @implNote Password length is preserved (which may be considered a security weakness, but is useful for
+     *           troubleshooting
+     *           and given state of Argo API's security... likely is an overkill already :)
+     * @return {@code ***}-masked string instead of the same length as configured password
+     */
+    public String getPasswordMasked() {
+        return this.password.replaceAll(".", "*");
```

Comment:
Check if it exists elsewhere (w/o regex)
```suggestion
        return this.password.replace(".", "*");
```

Reply:
Didn't find anything suitable in core (not saying it doesn't exist - it's just a testimony to my searching skills :) )
If you know of an alternative to switch to, please share.

Otherwise, I:
1. will demote it to private (to be less scary)
1. plan to leave as-is (as I'm [not the first](https://github.com/search?q=repo%3Aopenhab%2Fopenhab-addons%20password.replaceAll&type=code)) doing it this way

Reply:
Brainfart, it _is_ a regex, and not only replace dots.